tensorboard/webapp/widgets/linked_time_fob/linked_time_fob_controller_test.ts

```diff
@@ -401,5 +408,52 @@ describe('linked_time_fob_controller', () => {
         end: {step: 3},
       });
     });
+
+    it('changing start input modifies start step', () => {
+      let fixture = createComponent({
+        linkedTime: {start: {step: 1}, end: null},
+      });
+      fixture.detectChanges();
+
+      let mainDiv = fixture.debugElement.query(By.css('linked-time-fob div'));
+      mainDiv.triggerEventHandler('dblclick', {});
+      fixture.detectChanges();
+
+      let input = fixture.debugElement.query(By.css('linked-time-fob input'));
+
+      sendKeys(fixture, input, '8');
+      input.triggerEventHandler('change', {target: input.nativeElement});
+      fixture.detectChanges();
+
+      expect(onSelectTimeChanged).toHaveBeenCalledOnceWith({
+        start: {step: 8},
+        end: null,
+      });
+    });
+
+    it('changing end fob input modifies end step', () => {
+      let fixture = createComponent({
+        linkedTime: {start: {step: 1}, end: {step: 3}},
+      });
+      fixture.detectChanges();
+
+      let fobDivs = fixture.debugElement.queryAll(
```

Comment:
Similar to above: You should rearrange things so you can write something like:

const endFob = fixture.debugElement.query(By.css('linked-time-fob.endFob'));

const input = endFob.debugElement.query(By.css('input'));

Reply:
Done.